File: packages/square-mobile-payments/android/src/main/java/io/capawesome/capacitorjs/plugins/squaremobilepayments/classes/results/GetSettingsResult.java
package io.capawesome.capacitorjs.plugins.squaremobilepayments.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.squaremobilepayments.interfaces.Result;

public class GetSettingsResult implements Result {

    @NonNull
    private final String version;

    @NonNull
    private final String environment;

    public GetSettingsResult(@NonNull String version, @NonNull String environment) {
        this.version = version;
        this.environment = environment;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("version", version);
        result.put("environment", environment);
        return result;
    }
}

File: packages/square-mobile-payments/ios/Plugin/Classes/Results/GetSettingsResult.swift
import Foundation
import Capacitor

@objc public class GetSettingsResult: NSObject, Result {
    let version: String
    let environment: String

    init(version: String, environment: String) {
        self.version = version
        self.environment = environment
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["version"] = version
        result["environment"] = environment
        return result as AnyObject
    }
}
